src/sync/impl/applyRemote.js

```diff
@@ -249,8 +249,8 @@ export default function applyRemoteChanges(
       ...(_unsafeBatchPerCollection
         ? unsafeBatchesWithRecordsToApply(db, recordsToApply, sendCreatedAsUpdated, log)
         : [
-            db.batch(
-              ...prepareApplyAllRemoteChanges(db, recordsToApply, sendCreatedAsUpdated, log),
+            db.batchArray(
```

Comment:
explain - why would this have a significant impact on call stack size?

Reply:
Honestly, I still don’t understand why the error is about the size of the call stack, but this error appears because too many arguments are passed to the function.

Reply:
Ah! That makes sense, actually - it's not about the depth of the call stack, but the amount memory allocated to the stack.